### ISSUE TITLE
Feature request: BackgroundJob.ScheduleIfNotExists

### ISSUE THREAD

**Issue by alexey-pelykh** [HangfireIO/Hangfire#298]
Add ability to specify custom ID for BackgroundJob to exclude duplicates

**Comment by nirinchev** [comment]
This is a duplicate of https://github.com/HangfireIO/Hangfire/issues/181

**Comment by alexey-pelykh** [comment]
Indeed, but anyways it would be good to have some way to create "manual launch" jobs in Hangfire itself, instead of making additional entities to achieve that.

**Comment by karask** [comment]
I agree with @alexey-pelykh that this would be a simpler/cleaner way of doing this.

**Comment by edblackburn** [comment]
+1 I appreciate there are work arounds for achieve this behaviour but I think it would be better accomplished if it was first-class in he API.

**Comment by gandarez** [comment]
I've written a post about the new manual state job
https://discuss.hangfire.io/t/introducing-the-new-manual-state/1600
